packages/paypal-commerce-integration/src/paypal-commerce-alternative-methods/paypal-commerce-ratepay-payment-strategy.ts

```diff
@@ -194,6 +196,18 @@ export default class PaypalCommerceRatepayPaymentStrategy implements PaymentStra
         return `${date.getFullYear()}-${formattedMonth}-${formattedDate}`;
     }
 
+    private onPending(isLoading: boolean) {
+        const { onPending } = this.paypalcommerceratepay || {};
+
+        if (!onPending && typeof onPending !== 'function') {
+            throw new InvalidArgumentError(
+                `Unable to initialize payment because "options.getFieldsValues" argument is not provided.`,
+            );
+        }
+
+        onPending(isLoading);
+    }
```

Comment:
1. It might be better to make `onPending` callback optional;
2. I don't think that it is a right message for current implementation;
3. If the `onPending` is not provided, then `return` should be called;

Reply:
Please double check your if statement, when you will add new changes here

Reply:
made OnPending optional, if it not exist return from function